### PR TITLE
[#20150] feat: wallet connect pairing via deep links

### LINKED ISSUE #20150: Wallet Connect pairing via DeepLinks
# Feature Issue

[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to pair with the wallet via deeplinks, so that I can interact with the dApps I have on my phone.

[Link to User Story](https://www.notion.so/I-want-to-go-back-to-the-dApp-if-it-is-on-the-same-device-f4b4ff2e2bcf4985a92a837da12423d9?pvs=4)

### Description

[Docs](https://docs.walletconnect.com/web3wallet/mobileLinking) for deeplinking with Wallet Connect.

### Acceptance Criteria

- [ ] Wallet Connect pairing can be done via deeplinks (the connection screen appears after the user logs in)
- [ ] Wallet Connect signing requests can be done via deeplinks (the signing screen also appears after the login)
- [ ] The deeplinking works according to the flows described in the wallet connect docs (mentioned in the description of this issue)

### Notes

[comment]: # (For the developer who will implement. Include planning poker estimates, guidance, etc.)

### PR DESCRIPTION
**Pull request by mohsen-ghafouri**
fixes #20150

### Summary

As a user, I want to pair with the wallet via deeplinks, so that I can interact with the dApps I have on my phone.

#### Areas that maybe impacted

- Dapps connection with deep link

### Steps to test

- Open https://lab.web3modal.com/library/wagmi-siwe/
- Click the "Custom Wallet" button and add a custom name, in Mobile Linking type "status-app://"
- Click the "Add Wallet"
- Click the "Connect Wallet" button and choose the status wallet
- It should automatically open and redirect to your wallet.

### Result

https://github.com/user-attachments/assets/a49c8a93-794a-43d5-ac8a-ec02af974c66

status: ready

### CODE REVIEW

**Comment by mohsen-ghafouri** [comment]
I would appreciate any suggestion about how we can test app with installed dapps, in this website https://lab.web3modal.com/library/wagmi-siwe/ there is an option to add custom wallet but in example native app there is no such option.

**Comment by clauxx** [comment]
> I would appreciate any suggestion about how we can test app with installed dapps, in this website https://lab.web3modal.com/library/wagmi-siwe/ there is an option to add custom wallet but in example native app there is no such option.

We should submit it to the WalletConnect cloud so it's available in the explorer (https://walletconnect.com/explorer-guidelines). @jakubgs can we get access to the project to submit the app? Also, did we do this before with v1?

**Comment by mohsen-ghafouri** [comment]
@clauxx I can see old status app here https://explorer.walletconnect.com/status, the deeplink schema is the same(`status-app://`) we only need to update icon, and test it with release version maybe?

File: src/status_im/common/router.cljs
Comment (clauxx) [inline src/status_im/common/router.cljs]: can be a separate fn

File: src/status_im/contexts/wallet/wallet_connect/core.cljs
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/core.cljs]: can just be `first`, without `()`

File: src/status_im/contexts/wallet/wallet_connect/core.cljs
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/core.cljs]: can be just one `get-in` call

File: src/status_im/contexts/wallet/wallet_connect/responding_events.cljs
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/responding_events.cljs]: did you try it on iOS as well? I wonder what happens when we try to open the dapp.

File: src/status_im/contexts/wallet/wallet_connect/responding_events.cljs
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/responding_events.cljs]: can use `->>` to avoid unused assignments

**Review by clauxx — APPROVED** [review]
Have a few "stylistic" comments, but otherwise looks good!

File: src/status_im/contexts/wallet/wallet_connect/responding_events.cljs
Comment (mohsen-ghafouri) [reply src/status_im/contexts/wallet/wallet_connect/responding_events.cljs]: I first tested it with `url` and it was working fine in both platform but with `native` as i couldn't test it with installed app still don't know. 
I may create a simple dapp to test logic if it doesn't take time

Comment (clauxx) [reply src/status_im/contexts/wallet/wallet_connect/responding_events.cljs]: > I may create a simple dapp to test logic if it doesn't take time

This may be too time consuming I guess

Comment (mohsen-ghafouri) [reply src/status_im/contexts/wallet/wallet_connect/responding_events.cljs]: yeah i thought too, another option is to build app in release mode and test it with https://explorer.walletconnect.com/status, will let you know if it works fine

**Comment by jakubgs** [comment]
@clauxx yes, we have a WalletConnect project created over 2 years ago:

image

**Review by shivekkhurana — APPROVED** [review]

**Comment by clauxx** [comment]
> @clauxx yes, we have a WalletConnect project created over 2 years ago:
> 
> 
> 
> image
> 

Do you know what's with the draft tag below it?

**Comment by mohsen-ghafouri** [comment]
We need to have access to wallet connect to correct status app info, I build android in release mode and it didn't work with status configuration in wallet connect

File: src/status_im/contexts/wallet/wallet_connect/responding_events.cljs
Comment (mohsen-ghafouri) [reply src/status_im/contexts/wallet/wallet_connect/responding_events.cljs]: I removed the platform condition to have chance to test it with both platform, if in iOS it cause any issue i will disable it, in wallet connect doc it didn't mention any issue about iOS, lets see

**Comment by pavloburykh** [comment]
Hey @mohsen-ghafouri! Thanks for the PR. Could you please resolve the conflicts and rebase the branch? Thank you.

**Comment by clauxx** [comment]
> We need to have access to wallet connect to correct status app info, I build android in release mode and it didn't work with status configuration in wallet connect 

gotta check what's the bundle identifier in the cloud

**Comment by mohsen-ghafouri** [comment]
Hey @pavloburykh, thanks for checking this, done.

**Comment by pavloburykh** [comment]
@mohsen-ghafouri thanks for the PR. No issues from my side. Please, let us know once wallet connect team approves our new deep link configuration. After that we will re-check this feature using real dapps. Meanwhile I think we can merge this PR.

**Comment by mohsen-ghafouri** [comment]
Thanks @pavloburykh for you time, i will merge it once test pipeline being fixed.

**Pull request merged**
